### PR TITLE
adding dwb param descriptions

### PR DESCRIPTION
**Pull request by Marwan99**
per #21 

All DWB param descriptions should be there now. (hardest part completed). Examples to be added.
@SteveMacenski What do think of the layout? Any suggestions on where the publish and kinematic parameters should live?

### CODE REVIEW

File: configuration/packages/configuring-dwb-controller.rst [hunk 1]
```diff
@@ -0,0 +1,581 @@
+.. _configuring_dwb_controller:
+
+Configuring DWB Controller
```
Comment (SteveMacenski) [inline configuration/packages/configuring-dwb-controller.rst:3]:
```suggestion
DWB Controller
```

File: configuration/packages/configuring-dwb-controller.rst [hunk 2]
```diff
@@ -0,0 +1,581 @@
+.. _configuring_dwb_controller:
```
Comment (SteveMacenski) [inline configuration/packages/configuring-dwb-controller.rst:1]: I'm looking at your last CI job https://146-240847415-gh.circle-artifacts.com/0/html/configuration/index.html and it looks like this page is named "Configuring The Controller" and "Controller Server" but this section should only be DWB, nothing with the controller server. I don't know where this is.

Comment (Marwan99) [reply configuration/packages/configuring-dwb-controller.rst:1]: Not sure what was the issue, CI was re-triggered after the last commit and looks as expected now.
https://156-240847415-gh.circle-artifacts.com/0/html/index.html

let me know what you think of the layout, then I will go ahead and finalise it and fix the existing formatting issues.

Comment (SteveMacenski) [reply configuration/packages/configuring-dwb-controller.rst:1]: Weird ok

**Comment by SteveMacenski** [comment]
I think you should split the main page into 4 pages: dwb controller, iterator, kinematic parameters, visualization - along with the other separated pages. Its just too long. I think the main landing page here should just be a list of links to the other pages and then the full example below it. The DWB main-controller parameters should be first, then the other 3 on this main page, then the others you have listed.

File: configuration/packages/dwb-params/controller.rst [hunk 3]
```diff
@@ -1,11 +1,13 @@
-.. dwb_controller:
+.. _dwb_controller:
 
-DWB Controller
-==============
+Controller
```
Comment (SteveMacenski) [inline configuration/packages/dwb-params/controller.rst:3]:
```suggestion
DWB Controller
```

File: configuration/packages/dwb-params/iterator.rst [hunk 4]
```diff
@@ -1,11 +1,13 @@
-.. dwb_iterator:
+.. _dwb_iterator:
 
 Iterator
```
Comment (SteveMacenski) [inline configuration/packages/dwb-params/iterator.rst:3]: I think this is the XYTheta iterator, not just the generic iterator.

File: configuration/packages/dwb-params/kinematic.rst [hunk 5]
```diff
@@ -1,11 +1,13 @@
-.. dwb_kinematic_params:
+.. _dwb_kinematic_params:
 
 Kinematics
```
Comment (SteveMacenski) [inline configuration/packages/dwb-params/kinematic.rst:3]: this should also be the full name like the iterator page. Kinematic Parameters

File: configuration/packages/dwb-params/visualization.rst [hunk 6]
```diff
@@ -1,11 +1,13 @@
-.. dwb_vis:
+.. _dwb_vis:
 
 Visualization
```
Comment (SteveMacenski) [inline configuration/packages/dwb-params/visualization.rst:3]: As well here, use the full name

Comment (Marwan99) [reply configuration/packages/dwb-params/visualization.rst:3]: Full name for this is Publisher, right?

Comment (SteveMacenski) [reply configuration/packages/dwb-params/visualization.rst:3]: Correct

**Comment by SteveMacenski** [comment]
Reminder to self: post on discourse about this when this is in. Last PR before complete

**Comment by SteveMacenski** [comment]
@Marwan99 please  delete the `Tunable Parameters` page completely. We now have completely replaced it

**Comment by SteveMacenski** [comment]
Looks like you have a couple of merge conflicts the draft PR was hiding

**Comment by SteveMacenski** [comment]
Congrats @Marwan99! This was a big effort and we got it done in a couple weeks. What's your discourse handle?

**Comment by Marwan99** [comment]
Thanks @SteveMacenski, appreciate your guidance and support! My discourse handle is marwan99, just created the account!

**Comment by SteveMacenski** [comment]
https://discourse.ros.org/t/navigation2-complete-parameter-guide-now-available/14649

**Comment by SteveMacenski** [comment]
Celebrate the small victories or no one else will :-)

**Comment by Marwan99** [comment]
Should we tick the first item in this list? https://github.com/ros-planning/navigation2/issues/1589#issue-578188129

**Comment by SteveMacenski** [comment]
good call, done

**Pull request merged**